supabase/updateData.ts

```diff
@@ -81,6 +87,10 @@ export const saveData = async ({
 			last_message_id: topic.last_message_id ?? '',
 			message_count: topic.message_count ?? 0,
 			created_at: topic.thread_metadata?.create_timestamp ?? '',
+			updated_at:
+				newMessages.find((m) => m.id === topic.last_message_id)
+					?.edited_timestamp ??
+				newMessages.find((m) => m.id === topic.last_message_id)?.timestamp,
```

Comment:
You could lookup the last message before this and store it in a `const` to avoid finding it twice.